modules/email-template-service-impl/src/main/java/org/opencastproject/email/template/impl/EmailTemplateServiceImpl.java

```diff
@@ -130,25 +140,43 @@ public String applyTemplate(String templateName, String templateContent, Workflo
    */
   private HashMap<String, HashMap<String, String>> initCatalogs(MediaPackage mediaPackage, String delimiter) {
     HashMap<String, HashMap<String, String>> catalogs = new HashMap<String, HashMap<String, String>>();
-    Catalog[] dcs = mediaPackage.getCatalogs(DublinCoreCatalog.ANY_DUBLINCORE);
 
-    for (int i = 0; dcs != null && i < dcs.length; i++) {
+    Set<MediaPackageElementFlavor> catalogFlavors = new HashSet<>();
+    catalogFlavors.add(indexService.getCommonEventCatalogUIAdapter().getFlavor());
+    catalogFlavors.add(indexService.getCommonSeriesCatalogUIAdapter().getFlavor());
+    catalogFlavors.addAll(indexService.getEventCatalogUIAdapters().stream()
+        .map(CatalogUIAdapter::getFlavor)
+        .collect(Collectors.toSet()));
+    catalogFlavors.addAll(indexService.getSeriesCatalogUIAdapters().stream()
+        .map(CatalogUIAdapter::getFlavor)
+        .collect(Collectors.toSet()));
+
+    Set<Catalog> catalogElements = catalogFlavors.stream()
+        .flatMap(f -> Arrays.stream(mediaPackage.getCatalogs(f)))
+        .filter(Objects::nonNull)
+        .collect(Collectors.toSet());
+
+    for (Catalog c : catalogElements) {
       DublinCoreCatalog dc;
-      try (InputStream in = workspace.read(dcs[i].getURI())) {
+      try (InputStream in = workspace.read(c.getURI())) {
         dc = DublinCores.read(in);
       } catch (Exception e) {
         logger.warn("Error when populating catalog data", e);
         // Don't include the info
         continue;
       }
 
-      String catalogFlavor = dcs[i].getFlavor().getSubtype();
       HashMap<String, String> catalogHash = new HashMap<>();
       for (EName ename : dc.getProperties()) {
         String name = ename.getLocalName();
         catalogHash.put(name, dc.getAsText(ename, DublinCore.LANGUAGE_ANY, delimiter));
       }
-      catalogs.put(catalogFlavor, catalogHash);
+
+      catalogs.put(c.getFlavor().toString(), catalogHash);
+      // Backwards compatibility: use only subtype of the flavor as key
+      if (c.getFlavor().getType().equals("dublincore")) {
+        catalogs.put(c.getFlavor().getSubtype(), catalogHash);
+      }
```

Comment:
IMO not mentioning it is better.  We want to do away with the old style ideally, so not mentioning it would encourage people to upgrade.

Reply:
If you want to encourage people to upgrade, it's better to fail fast/early and not have backward compatibility instead of having undocumented behavior. It's still working, but nobody knows why.
You usually document that a method is deprecated, and a few releases later, you remove the stuff.

Reply:
I agree with @geichelberger and added a notice to the docs.